home/views.py

```diff
@@ -117,36 +119,44 @@ def get_context_data(self, **kwargs):
 
 
 class ContentUploadThread(threading.Thread):
-    def __init__(self, file, splitlines, newline, purge, locale, **kwargs):
+    def __init__(self, file, file_type, purge, locale, **kwargs):
         self.file = file
-        self.splitlines = splitlines
+        self.file_type = file_type
         self.purge = purge
         self.locale = locale
-        self.newline = newline
         super(ContentUploadThread, self).__init__(**kwargs)
 
     def run(self):
-        import_content_csv(
-            self.file, self.splitlines, self.newline, self.purge, self.locale
-        )
+        import_content(self.file, self.file_type, self.purge, self.locale)
+
+
+class UploadView(View):
+    form_class = UploadFileForm
+    template_name = "upload.html"
 
+    def get(self, request, *args, **kwargs):
+        loading = "ContentUploadThread" in [th.name for th in threading.enumerate()]
```

Comment:
The issue with using threads like this, is that we're assuming we'll only ever have a single process running. If we have multiple processes, then we're not guaranteed that the http request will always go to the same one.

I think we can leave it for now, but for future we might want to think about storing this state in the database instead.

Reply:
but they won't be able to access the import page through the UI if while there is a process running it disables the form?

Reply:
Say you've scaled your webserver up, say to two processes, A and B. You make a web request to start the import, it goes to A, which starts a thread on A. Then the browser makes another request to check the status of the import, this then gets send to B. B doesn't have any threads running, so `loading` is `False`, so the page redirects, even though the import task is still running on A.

By default gunicorn has a single sync worker, so this will work. But if you need to support a higher number of requests, you might want to increase the amount of workers, or even increase the number of instances (docker containers). Since there's no communication between these, they might even be on different machines, you'll want a central place to store this state. Database is good because we don't need to add another dependancy, but this could also be a shared cache like Redis.

Reply:
ah I see. Adding that as an Issue on GitHub so we don't forget